File: frontend/customer/mobile/bereload_FE/lib/screens/home/services/sim_swap_service.dart
class SimSwapService {
  static final SimSwapService _instance = SimSwapService._internal();
  factory SimSwapService() => _instance;
  SimSwapService._internal();

  final List<Map<String, String>> requests = [
    {
      'date': '25 Jan 2025',
      'status': 'Completed',
      'oldNumber': '+32493454910',
      'newNumber': '+32494590891',
      'reason': 'Network issue',
    }
  ];

  void addRequest(Map<String, String> request) {
    requests.insert(0, request);
  }
}

File: frontend/customer/web/bereload/src/services/sim-swap-service.ts
export class SimSwapService {
    private static instance: SimSwapService;
    
    private _requests: Array<{
      date: string;
      status: string;
      oldNumber: string;
      newNumber: string;
      reason: string;
    }> = [
      {
        date: '25 Jan 2025',
        status: 'Completed',
        oldNumber: '+32493454910',
        newNumber: '+32494590891',
        reason: 'Network issue',
      }
    ];
  
    private constructor() {}
  
    public static getInstance(): SimSwapService {
      if (!SimSwapService.instance) {
        SimSwapService.instance = new SimSwapService();
      }
      return SimSwapService.instance;
    }
  
    get requests() {
      return this._requests;
    }
  
    addRequest(request: {
      date: string;
      status: string;
      oldNumber: string;
      newNumber: string;
      reason: string;
    }) {
      this._requests.unshift(request); 
    }
  }
  
  export function formatCurrentDate(): string {
    const now = new Date();
    const day = String(now.getDate()).padStart(2, '0');
    const months = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'];
    const month = months[now.getMonth()];
    const year = now.getFullYear();
    
    return `${day} ${month} ${year}`;
  }
  
  export default SimSwapService.getInstance();